ObservableCollection/MintPlayer.ObservableCollection/ObservableCollection.cs

```diff
@@ -16,102 +16,147 @@
 
-namespace MintPlayer.ObservableCollection
+namespace MintPlayer.ObservableCollection;
+
+public class ObservableCollection
+    <[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces | DynamicallyAccessedMemberTypes.PublicMethods)] T> 
+    : System.Collections.ObjectModel.ObservableCollection<T>, IDisposable
 {
-    public class ObservableCollection<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] T> : System.Collections.ObjectModel.ObservableCollection<T>
+    #region Constructors
+    public ObservableCollection()
     {
-        #region Constructors
-        public ObservableCollection()
-        {
-        }
+        areItemsImplementingINotifyPropertyChanged = typeof(T).GetInterfaces().Contains(typeof(INotifyPropertyChanged));
+    }
 
-        public ObservableCollection(IEnumerable<T> items) : this()
+    public ObservableCollection(IEnumerable<T> items) : this()
+    {
+        InternalAddRange(items);
+        if (areItemsImplementingINotifyPropertyChanged)
         {
-            InternalAddRange(items);
+            var notifyItems = Items.Cast<INotifyPropertyChanged>();
+            itemNotifyPropertyChangedList.AddRange(notifyItems);
+            foreach (var item in notifyItems)
+            {
+                item.PropertyChanged += ObservableCollection_Item_PropertyChanged;
+            }
         }
-        #endregion Constructors
+    }
+    #endregion Constructors
 
-        #region Private fields
+    #region Finalizer
 
-        private bool isAddingOrRemovingRange;
-        [NonSerialized] private DeferredEventsCollection? deferredEvents;
-        private readonly SynchronizationContext? synchronizationContext = SynchronizationContext.Current;
+    ~ObservableCollection()
+    {
+        Dispose(false);
+    }
 
-        #endregion Private fields
+    #endregion
 
-        #region Public methods
+    #region Private fields
 
-        public virtual void AddRange(IEnumerable<T> items)
-        {
-            CheckReentrancy();
+    private bool isAddingOrRemovingRange;
+    [NonSerialized] private DeferredEventsCollection? deferredEvents;
+    private readonly SynchronizationContext? synchronizationContext = SynchronizationContext.Current;
 
-            RunOnMainThread((param) =>
-            {
-                InternalAddRange(param.items);
-                OnCountPropertyChanged();
-                OnIndexerPropertyChanged();
-                OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Add, param.items));
-            }, new { items = items.ToArray() });
+    private readonly bool areItemsImplementingINotifyPropertyChanged;
+    private readonly List<INotifyPropertyChanged> itemNotifyPropertyChangedList = [];
+    private bool _disposed;
+
+    #endregion Private fields
+
+    #region Public methods
+
+    public virtual void AddRange(IEnumerable<T> items)
+    {
+        var enumerable = items as T[] ?? items.ToArray();
```

Comment:
Is this necessary? I know a single item is not a range-operation, but we might as well deal with it the same way?